plugins/nf-azure/src/main/nextflow/cloud/azure/nio/AzFileSystem.groovy

```diff
@@ -408,8 +408,15 @@ class AzFileSystem extends FileSystem {
 
     @PackageScope
     void copy(AzPath source, AzPath target) {
+        final sasToken = provider.getSasToken()
+        String sourceUrl = source.blobClient().getBlobUrl()
+
+        if (sasToken != null) {
+            sourceUrl += "?${sasToken}"
+        }
```

Comment:
Looking the implementation of `getBlobUrl` the url can contains parameters, therefore appending `?` is a bt tool optimistic  

```
    public String getBlobUrl() {
        String blobUrl = azureBlobStorage.getUrl();
        if (this.isSnapshot()) {
            blobUrl = Utility.appendQueryParameter(blobUrl, "snapshot", getSnapshotId());
        }
        if (this.getVersionId() != null) {
            blobUrl = Utility.appendQueryParameter(blobUrl, "versionid", getVersionId());
        }
        return blobUrl;
    }
```

The `Utility.appendQueryParameter` method should be used instead

Reply:
I could not use Utility.appendQueryParameter since that needs a key and the sasToken does not have any but I took possible url parameters into account